Backends/Wayland/README.org

```diff
@@ -0,0 +1,146 @@
+* Wayland Backend (warning: work-in-progress quality!)
+
+This is currently in a proof-of-concept state for getting some early code review. I'm currently learning both CLIM, McCLIM, and Wayland so I'm sure I'm "doing it wrong" in quite a few places.
+
+I'm testing this on a GUIX OS using sway as my wayland compositor. I haven't yet tested in a GNOME or KDE contexts at all.
+
+I plan to rebase this branch again McCLIM/master frequently as I work.
+
+* What "works"
+1. A visual window can been seen on the screen
+2. It resizes in sway as it's moved around
+3. The sway keychord for quitting an app (s-M-q on mine) kinda works, but not without error.
+4. =destroy-port= works
+5. Some keyboard and pointer... at least debug logging -- untested in a CLIM app.
+6. Basic rendering of window elements e.g. scroll bars
+
+* What doesn't work as of 2022-11-19
+1. Everything else :) TODO list now being maintained below.
+
+
+* Running
+
+** Dependencies
+This currently requires my feature branch of Andrew Soutar's great =cl-wayland-client= package which adds extra FFI bindings for EGL and convenience path search for finding the wayland protocol libraries. Neither are on quicklisp currently. I'll be seeing if upstreaming is a possibility but I haven't testing for non-breaking on more mainstream distros.
+
+I realize this is a weakness.
+
+** Environment
+You'll need development library for wayland and mesa. For debian/ubuntu based distributions, it usually means the =*-dev= packages. On Guix, I'm doing this in a dedicated dev environment:
+#+begin_src sh
+guix shell sbcl mesa wayland wayland-protocols libxkbcommon
+#+end_src
+
+I needed to start emacs in a way it could find the wayland protocal specs and libraries. You might not need this if you aren't sandboxing your environment:
+#+begin_src sh
+LD_LIBRARY_PATH="$GUIX_ENVIRONMENT/lib" emacs &
```

Comment:
I think setting LD_LIBRARY_PATH on Guix is considered to be a hack. Guix binds $LIBRARY_PATH to where the libs are symlinked. You could put something like this in your sbclrc (or do something more fancy with LIBRARY_PATH):
```
(ql:quickload :cffi)
(push (format nil "~a/.guix-profile/lib/" (uiop:getenv "HOME")) cffi:*foreign-library-directories*)
```

Reply:
Thanks. Yes, I agree it's mostly a quick one-off solution for now. If it is a "hack," it's been an extremely useful one for me in the past when debugging OpenGL calls in lisp with gDebugger circa 2006-8 or so.

I'll give it a try soon, though.